Client/Frontend/Browser/DefaultBrowserIntroManager.swift

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+import BraveShared
+
+private typealias IntroPrefs = Preferences.DefaultBrowserIntro
+
+struct DefaultBrowserIntroManager {
+    /// This function should be called when the app is initialized.
+    /// It determines whether we should show default browser intro popup or not and sets corresponding preferences accordingly.
+    /// Returns true if the popup should be shown.
+    @discardableResult
+    static func prepareAndShowIfNeeded(isNewUser: Bool, launchDate: Date = Date()) -> Bool {
+        // Empty new user value means it is the first time this code is called after
+        // the default intro callout was introduced. Afterwards it should be never nil.
+        if IntroPrefs.isNewUser.value == nil {
```

Comment:
I really didnt understand why we need IntroPrefs.isNewUser, we already have if it is a first launch or not information.